gn3/db_utils.py

```diff
@@ -24,8 +25,10 @@ def cursor(self, *args, **kwargs) -> Any:
 
 
 @contextlib.contextmanager
-def database_connection(sql_uri) -> Iterator[Connection]:
+def database_connection(sql_uri: str, logger: Optional[logging.Logger] = None) -> Iterator[Connection]:
     """Connect to MySQL database."""
+    if logger is None:
```

Comment:
Hey @jnduli 

*suggestion*: Why not move the `logger = logging.getLogger(...)` out of this function, to the module's top level, then pass it as the default logger. That way you can get rid of the `if logger is None` check and the logger is also available to any other function that might need logging in the module.

Reply:
Thanks. This makes the implementation much cleaner. I've pushed the suggestion.